components/tikv_util/src/timer.rs

```diff
@@ -196,28 +185,88 @@ impl Default for SteadyTimer {
 }
 
 fn start_global_steady_timer() -> SteadyTimer {
-    let (tx, rx) = mpsc::channel();
     let clock = SteadyClock::default();
-    let clock_ = clock.clone();
+    let handle = start_timer_thread("steady-timer", clock.clone());
+    SteadyTimer { clock, handle }
+}
+
+/// A clock that ratchets forward.
+///
+/// It is used to workaround a panic[^1] in tokio-timer when the clock goes
+/// backward, which is possible in some environments, e.g., aliyun ECS g7.
+///
+/// [^1]: https://github.com/tokio-rs/tokio/pull/515
+struct RatchetClock<T: Now> {
+    // Although the RatchetClock has to be thread-safe (Now requires Sync+Send),
+    // the [`tokio_timer::timer::Clock`] that wraps it can only be accessed by
+    // the timer thread. Therefore, performance penalty of using a Mutex is negligible.
+    state: Mutex<RatchetClockState<T>>,
```

Comment:
While I still think a `Cell<u64>` is enough here, using a Mutex here is acceptable to me as there should be no performance impact.